lib/saluki-env/src/workload/providers/remote_agent/api.rs

```diff
@@ -41,44 +49,34 @@ impl RemoteAgentWorkloadState {
             active_entities.insert(entity_id.clone());
         });
 
-        self.tag_querier.visit_entity_mappings(|entity_id, parent_id| {
+        self.tag_querier.visit_entity_aliases(|entity_id, target_entity_id| {
             active_entities.insert(entity_id.clone());
-            entity_mappings.insert(entity_id.clone(), parent_id.clone());
+            entity_aliases.insert(entity_id.clone(), target_entity_id.clone());
         });
 
-        // For each entity, build its ancestry chain and collect all of the relevant tags.
+        // For each entity, build its information.
         for entity_id in active_entities.iter() {
-            let mut ancestors = Vec::new();
-
-            let mut current_entity = entity_id;
-            while let Some(parent) = entity_mappings.get(current_entity) {
-                ancestors.insert(0, parent);
-                current_entity = parent;
-            }
-
+            let alias = entity_aliases.get(entity_id);
             let low_cardinality_tags = self
                 .tag_querier
-                .get_entity_tags(entity_id, OriginTagCardinality::Low)
+                .get_unaliased_entity_tags(entity_id, OriginTagCardinality::Low)
```

Comment:
Why `.get_unaliased_entity_tags` here? Doesn't the old implementation using `TagSnapshot` have the unified tags so the dump would show those tags for the `entity_id` but here we are leaving it empty if we have something that has an alias.